src/hera/workflows/_mixins.py

```diff
@@ -1128,6 +1225,26 @@ def _get_params_from_items(with_items: List[Any]) -> Optional[List[Parameter]]:
     return [Parameter(name=n, value=f"{{{{item.{n}}}}}") for n in with_items[0].keys()]
 
 
+def _process_params_and_artifacts(params, artifacts):
+    if len(params) == 1:
+        params = [Parameter(name=p.name, value="{{item}}") for p in params]
```

Comment:
The `for p in params` looks strange/implies more items, even though we have `len(params) == 1`

Reply:
True, I took it from somewhere else in the _mixins code